cmd/commands/start.go

```diff
@@ -127,11 +129,15 @@ func startTendermint(dir string, basecoinApp *app.Basecoin) error {
 	}
 	cfg.SetRoot(cfg.RootDir)
 	config.EnsureRoot(cfg.RootDir)
-	logger.SetLogLevel(cfg.LogLevel)
+
+	tmLogger, err := log.NewFilterByLevel(logger, cfg.LogLevel)
+	if err != nil {
+		return err
+	}
 
 	// Create & start tendermint node
-	privValidator := types.LoadOrGenPrivValidator(cfg.PrivValidatorFile())
-	n := node.NewNode(cfg, privValidator, proxy.NewLocalClientCreator(basecoinApp))
+	privValidator := types.LoadOrGenPrivValidator(cfg.PrivValidatorFile(), tmLogger)
+	n := node.NewNode(cfg, privValidator, proxy.NewLocalClientCreator(basecoinApp), tmLogger.With("module", "node"))
```

Comment:
I'm wondering if these `With` calls should be the responsibility of the function the logger is being passed to, rather than the caller ....